src/app/container/tool-file-editor/tool-file-editor.component.ts

```diff
@@ -66,17 +72,20 @@ export class ToolFileEditorComponent extends FileEditing {
    * Splits up the sourcefiles for the version into descriptor files and test parameter files
    */
   loadVersionSourcefiles(): void {
-    this.descriptorFiles = JSON.parse(JSON.stringify(this.getDescriptorFiles(this.currentVersion.sourceFiles)));
-    this.testParameterFiles = JSON.parse(JSON.stringify(this.getTestFiles(this.currentVersion.sourceFiles)));
-    this.dockerFile = JSON.parse(JSON.stringify(this.getDockerFile(this.currentVersion.sourceFiles)));
+    this.containerTagsService.getTagsSourcefiles(this.id, this.currentVersion.id).subscribe((sourcefiles: Array<SourceFile>) => {
+      this.originalSourceFiles = sourcefiles;
+      this.descriptorFiles = JSON.parse(JSON.stringify(this.getDescriptorFiles(this.originalSourceFiles)));
```

Comment:
May be missing something, it seems like you have JSON, which you stringify, then convert back to JSON. Can you just use the result directly, with stringifying it and parsing it?

Reply:
Not sure about this case specifically. But for other areas where you see this stringify + parse, it's because it's a leftover issue with Akita integration.  Akita has readonly objects but we sometimes use it as-is with something like ngModel which will not work.

Reply:
Ah, so I was missing something. A comment explaining why we're doing this might be useful. Or extract it into a method `cloneJSONForAkita` or something like that... But perhaps beyond the scope of this PR.

Reply:
Still feel this deserves a comment.